pkg/api/types.go

```diff
@@ -689,6 +689,10 @@ type TestStepConfiguration struct {
 	// Timeout overrides maximum prowjob duration
 	Timeout *prowv1.Duration `json:"timeout,omitempty"`
 
+	// DisableRehearsal will exclude the "can-be-rehearsed" label from the pj spec.
+	// Which will mean the test will not be picked up for rehearsals. Not valid on postsubmits.
+	DisableRehearsal bool `json:"disable_rehearsal,omitempty"`
```

Comment:
I think it would be better not to interact with `ci-operator` at all. We can use the prowgen configuration file, to allow users to disable the rehearsal in specific jobs.

Reply:
Check: https://github.com/openshift/ci-tools/blob/master/pkg/config/load.go#L21-L34

Reply:
That's what I am doing in https://github.com/openshift/ci-tools/pull/2979, but I liked the idea of making them configurable at the job level. With the prowgen config it's all or nothing

Reply:
We can configure them at the job level in the prowgen configuration too. Something like
```yaml
exclude_rehearsals:
- test-1
- test-2
- test-3
```

Reply:
Right, that's what I would like to avoid doing though. That would leave us with test names in two separate files. We would have to keep the files in sync and create an entirely new type of validation.

Can you explain the downsides to adding a new Boolean field where I have done it here?

Reply:
My concern is that this is a temporary workaround until we change the rehearsal tool to a prow external plugin. Also, the rehearsals have nothing to do with ci-operator, so I think it would be better not to include this logic in the ci-operator config. 
Prowgen is responsible for generating the job and anything related to it should be included in prowgen config only. 

Also, following this logic, we should remove completely the prowgen config and include all of those options in ci-operator config, but this will mess things up. 

I believe not messing up with ci-operator config is the best option here, since this is a temporary workaround.